client/signup/main.jsx

```diff
@@ -88,23 +92,21 @@ class Signup extends React.Component {
 
 	constructor( props, context ) {
 		super( props, context );
-		SignupDependencyStore.setReduxStore( context.store );
 
 		this.state = {
+			controllerHasReset: false,
 			login: false,
-			progress: SignupProgressStore.get(),
 			dependencies: props.signupDependencies,
 			loadingScreenStartTime: undefined,
 			resumingStep: undefined,
 			loginHandler: null,
 			hasCartItems: false,
 			plans: false,
 			previousFlowName: null,
-			resumedAfterLogin: false,
 		};
 	}
 
-	componentWillMount() {
+	UNSAFE_componentWillMount() {
```

Comment:
can we move this to DidMount or the constructor instead of moving to UNSAFE_*?

Reply:
Given the large and interconnected nature of the main signup script, I wanted to avoid introducing more changes than necessary with this PR.

It's certainly worth considering once we land this change, though.